packages/database/src/migrations/20210706044733-SwitchLGALevelMapOverlayToUseReportServer-modifies-data.js

```diff
@@ -0,0 +1,222 @@
+    denominatorDataElementCode: 'FWV_004',
+    numeratorTitle: 'Participants with symptoms tested for flu',
+  },
+  {
+    id: 'AU_FLUTRACKING_Tested_For_Covid',
+    numeratorDataElementCode: 'FWV_010',
+    denominatorDataElementCode: 'FWV_004',
+    numeratorTitle: 'Participants with symptoms tested for covid',
+  },
+  {
+    id: 'AU_FLUTRACKING_Tested_Positive_For_Flu',
+    numeratorDataElementCode: 'FWV_011',
+    denominatorDataElementCode: 'FWV_004',
+    numeratorTitle: 'Participants with symptoms tested positive for flu',
+  },
+  {
+    id: 'AU_FLUTRACKING_Tested_Positive_For_Covid',
+    numeratorDataElementCode: 'FWV_012',
+    denominatorDataElementCode: 'FWV_004',
+    numeratorTitle: 'Participants with symptoms tested positive for covid',
+  },
+];
+
+const getReport = ({
+  code,
+  numeratorDataElementCode,
+  denominatorDataElementCode,
+  numeratorTitle,
+}) => {
+  const numTitle = `'${numeratorTitle}'`;
+  const denTitle = `'${denominatorTitle}'`;
+  return {
+    code,
+    config: {
+      fetch: {
+        aggregations: ['MOST_RECENT'],
+        dataElements: [numeratorDataElementCode, denominatorDataElementCode],
+      },
+      transform: [
+        {
+          transform: 'aggregate',
+          dataElement: 'group',
+          organisationUnit: 'group',
+          value: 'sum',
+          '...': 'drop',
+        },
+        'keyValueByDataElementName',
+        {
+          transform: 'aggregate',
+          organisationUnit: 'group',
+          '...': 'sum',
+        },
+        {
+          transform: 'filter',
+          where: `exists($row.${numeratorDataElementCode}') and exists($row.${denominatorDataElementCode})`,
+        },
+        {
+          transform: 'select',
+          "'value'": `divide($row.${numeratorDataElementCode},$row.${denominatorDataElementCode})`,
+          [numTitle]: `round($row.${numeratorDataElementCode})`,
+          [denTitle]: `round($row.${denominatorDataElementCode})`,
+          "'organisationUnitCode'": '$row.organisationUnit',
+        },
+      ],
+    },
+  };
+};
+
+exports.up = async function (db) {
+  const permissionGroupId = (await findSingleRecord(db, 'permission_group', { name: 'Public' })).id;
+  for (const mapOverlay of mapOverlays) {
+    const { id, numeratorDataElementCode, denominatorDataElementCode, numeratorTitle } = mapOverlay;
+    await insertObject(db, 'report', {
+      ...getReport({
+        code: id,
+        numeratorDataElementCode,
+        denominatorDataElementCode,
+        numeratorTitle,
+      }),
+      id: generateId(),
+      permission_group_id: permissionGroupId,
+    });
+
+    await updateValues(
+      db,
+      'mapOverlay',
+      {
+        measureBuilder: 'useReportServer',
+        measureBuilderConfig: { reportCode: id, dataSourceType: 'custom' },
+      },
+      { id },
+    );
+  }
+
+  await db.runSql(`
+    UPDATE "mapOverlay"
+    SET "presentationOptions" = "presentationOptions" || '{"notLegacy": true}'::jsonb
```

Comment:
It will be nice to have a `legacy` column for `mapOverlay` to be consistent to `dashboard`.